src/main/java/fr/sncf/osrd/train/ActivateRoute.java

```diff
@@ -9,44 +9,47 @@
-        var nextRouteState = sim.infraState.getRouteState(nextRoute.index);
-        // Try to reserve the route if possible
-        if (nextRouteState.status == RouteStatus.FREE)
-            nextRouteState.reserve(sim);
+    public static void reserveRoutes(Simulation sim, SignalNavigatePhase.State navigatePhaseState, Train train)
+            throws SimulationError {
+        for (int advance = 1; advance <= 1; advance++) {
+            if (navigatePhaseState.getRouteIndex() + advance < navigatePhaseState.phase.routePath.size()) {
+                var nextRoute = navigatePhaseState.phase.routePath.get(navigatePhaseState.getRouteIndex() + advance);
+                var nextRouteState = sim.infraState.getRouteState(nextRoute.index);
+                sim.infraState.switchPost.request(sim, nextRouteState, train);
+            }
+        }
     }
 
-    /** Reserve the initial routes, mark occupied tvd sections and add interactable elements that are under the train
-     * to the TrainState*/
+    /**
+     * Reserve the initial routes, mark occupied tvd sections and add interactable
+     * elements that are under the train to the TrainState
+     */
     public static void trainCreation(Simulation sim, TrainState trainState) throws SimulationError {
         Route route = trainState.trainSchedule.initialRoute;
         var routeState = sim.infraState.getRouteState(route.index);
 
         // Reserve the initial route
         if (routeState.status != RouteStatus.FREE)
-            throw new SimulationError(String.format(
-                    "Impossible to reserve the route '%s' since it is not available.", routeState.route.id));
+            throw new SimulationError(String.format("Impossible to reserve the route '%s' since it is not available.",
+                    routeState.route.id));
         routeState.initialReserve(sim);
 
         // Reserve the tvdSection where the train is created
         var trainPosition = trainState.location.trackSectionRanges.getFirst();
 
+        var lastTvdSectionPath = route.tvdSectionsPaths.get(0);
+        occupyTvdSectionPath(sim, lastTvdSectionPath);
+
         for (var i = 0; i < route.tvdSectionsPaths.size(); i++) {
             var currentTvdSectionPath = route.tvdSectionsPaths.get(i);
-            occupyTvdSectionPath(sim, currentTvdSectionPath);
             var currentTvdSectionPathDirection = route.tvdSectionsPathDirections.get(i);
             for (var trackSection : currentTvdSectionPath.getTrackSections(currentTvdSectionPathDirection)) {
                 if (trainPosition.intersect(trackSection))
                     return;
             }
-            freeTvdSectionPath(sim, currentTvdSectionPath);
+            freeTvdSectionPath(sim, lastTvdSectionPath);
+            occupyTvdSectionPath(sim, currentTvdSectionPath);
+            lastTvdSectionPath = currentTvdSectionPath;
         }
     }
```

Comment:
@flomonster I think you understand this method more than I do, does it make sense?